hoot-core/src/main/cpp/hoot/core/conflate/CookieCutter.cpp

```diff
@@ -32,58 +32,81 @@
 #include <hoot/core/util/MapProjector.h>
 #include <hoot/core/ops/SuperfluousNodeRemover.h>
 #include <hoot/core/ops/SuperfluousWayRemover.h>
-#include <hoot/core/ops/MapCropper.h>
 #include <hoot/core/util/GeometryUtils.h>
 #include <hoot/core/visitors/UnionPolygonsVisitor.h>
 #include <hoot/core/visitors/CalculateMapBoundsVisitor.h>
+#include <hoot/core/io/OsmMapWriterFactory.h>
 #include <hoot/core/util/Log.h>
+#include <hoot/core/ops/MapCropper.h>
 
 using namespace geos::geom;
 
 namespace hoot
 {
 
-CookieCutter::CookieCutter(bool crop, double outputBuffer) :
+CookieCutter::CookieCutter(bool crop, double outputBuffer, bool keepEntireFeaturesCrossingBounds,
+                           bool keepOnlyFeaturesInsideBounds) :
```

Comment:
Mostly added a lot of debugging code and more comments to this class. The only functionality that got added were setters for some of the new crop options.